dataquality/loggers/data_logger/seq2seq.py

```diff
@@ -0,0 +1,188 @@
+        Ex: Billsum dataset, with `text` input and `summary` as the label
+        id  text	                        summary
+        0	SECTION 1. LIABILITY ...	    Shields a business entity ...
+        1	SECTION 1. SHORT TITLE.\n\n ...	Human Rights Information Act ...
+        2	SECTION 1. SHORT TITLE.\n\n ...	Jackie Robinson Commemorative Coin ...
+        3	SECTION 1. NONRECOGNITION ...	Amends the Internal Revenue Code to ...
+        4	SECTION 1. SHORT TITLE.\n\n ...	Native American Energy Act - (Sec. 3...
+
+        You can log your dataset via the `dq.log_dataset` function, passing in the
+        column mapping as necessary for `text`, `label`, and `id`
+        `dq.log_dataset(ds, text="text", label="summary", id="id")`
+
+    Putting it all together:
+        from datasets import load_dataset
+        from transformers import T5TokenizerFast
+
+        tokenizer = T5TokenizerFast.from_pretrained("t5-small")
+        ds = load_dataset("billsum")
+        # Add `id` column to each dataset split as the idx
+        ds = ds.map(lambda x,idx : {"id":idx},with_indices=True)
+        dq.init("seq2seq")
+        dq.set_tokenizer(tokenizer)
+        dq.log_dataset(ds["train"], label="summary", split="train")
+
+    NOTE: We assume that the tokenizer you provide is the same tokenizer used for
+    training. This must be true in order to align inputs and outputs correctly. Ensure
+    all necessary properties (like `add_eos_token`) are set before setting your
+    tokenizer so as to match the tokenization process to your training process.
+    """
+
+    __logger_name__ = "seq2seq"
+    logger_config = seq2seq_logger_config
+
+    def __init__(self, meta: Optional[MetasType] = None) -> None:
+        super().__init__(meta)
+        # Tokens IDs in a given input string
+        self.tokenized_labels: List[List[int]] = []
+        # Character offsets for each token (from tokenized_inputs) in the dataset
+        self.token_label_offsets: List[List[Tuple[int, int]]] = []
+        # Index (or indices) into the token array for every offset
+        self.token_label_positions: List[List[Set[int]]] = []
+        self.ids: List[int] = []
+        self.texts: List[str] = []
+        self.labels: List[str] = []
+
+    @property
+    def token_map_key(self) -> str:
+        if self.split == Split.inference and self.inference_name is not None:
+            return self.inference_name
+        return str(self.split)
+
+    def validate_and_format(self) -> None:
+        super().validate_and_format()
+        label_len = len(self.labels)
+        text_len = len(self.texts)
+        id_len = len(self.ids)
+        assert id_len == text_len == label_len, (
+            "IDs, text, and labels must be the same length, got "
+            f"(ids, text, labels) ({id_len}, {text_len}, {label_len})"
+        )
+        assert (
+            self.logger_config.tokenizer is not None
+        ), "You must set your tokenizer before logging. Use `dq.set_tokenizer`"
```

Comment:
Not formatted to 88, but the below should be fine.

```suggestion
        assert self.logger_config.tokenizer, "You must set your tokenizer before logging. Use `dq.set_tokenizer`"
```

Reply:
ill update it locally 👍